wsd/COOLWSD.cpp

```diff
@@ -1001,6 +1007,111 @@ class PrisonPoll : public TerminatingPoll
 /// And also cleans up and balances the correct number of children.
 static std::unique_ptr<PrisonPoll> PrisonerPoll;
 
+#ifdef __linux__
+#if !MOBILEAPP
+class InotifySocket : public Socket
+{
+public:
+    InotifySocket():
+        Socket(inotify_init1(IN_NONBLOCK))
+        , m_stopOnConfigChange(true)
+    {
+        if (getFD() == -1)
+        {
+            LOG_WRN("Inotify - Failed to start a watcher for the configuration, disabling "
+                    "stop_on_config_change");
+            m_stopOnConfigChange = false;
+            return;
+        }
+
+        watch("/etc/coolwsd/coolwsd.xml");
```

Comment:
COOLWSD_CONFIGDIR is the path of coolwsd.xml. 
What about watching COOLWSD_CONFIGDIR itself? That way we could cover coolkitconfig.xcu as well, and at the same time we would not have to worry about strange behaviour of vim. Would there be any side-effects?

Reply:
Agreed - would prefer watching the parent directory and then matching filenames with changed size / attributes etc.